CMakeLists.txt

```diff
@@ -258,14 +258,19 @@ if (NOT CMAKE_BUILD_TYPE)
 endif (NOT CMAKE_BUILD_TYPE)
 
 if (NOT (WIN32 OR CYGWIN))
-  option (BUILD_SHARED_LIBS "Build shared libraries." ON)
+  option (SYSTEMC_BUILD_SHARED_LIBS "Build shared libraries." ON)
 else (NOT (WIN32 OR CYGWIN))
-  option (BUILD_SHARED_LIBS "Build shared libraries." OFF)
+  option (SYSTEMC_BUILD_SHARED_LIBS "Build shared libraries." OFF)
 endif (NOT (WIN32 OR CYGWIN))
-if (BUILD_SHARED_LIBS AND (WIN32 OR CYGWIN))
-  message (WARNING "The compilation of SystemC as a DLL on Windows is currently not supported!")
-  set (BUILD_SHARED_LIBS CACHE BOOL "Build shared libraries." OFF FORCE)
-endif (BUILD_SHARED_LIBS AND (WIN32 OR CYGWIN))
+
+set(SC_LIB_TYPE STATIC)
+if (SYSTEMC_BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+  if (WIN32 OR CYGWIN)
+    message (FATAL_ERROR "The compilation of SystemC as a DLL on Windows is currently not supported!")
+  else ()
```

Comment:
I think that this is not anymore valid, as the infrastructure for the DLL exports is in place. However, I never built DLLs on Windows, as I am primarily using Linux and macOS.

@pah: Can you comment on this?